### PR TITLE
[docs] Replaced `&#151;` with `---` in remat and checkpointing docs

### PR DESCRIPTION
**Pull request by vfdev-5**
Replaced

image

into

image

https://jax--31809.org.readthedocs.build/en/31809/notebooks/autodiff_remat.html

### CODE REVIEW

**Comment by hawkinsp** [comment]
Is it possible to change it somehow to use an em-dash, as it was originally intended?

**Comment by hawkinsp** [comment]
Nit: I suspect that's an en-dash, not an em-dash. But close enough.

**Review by hawkinsp — APPROVED** [review]

**Comment by vfdev-5** [comment]
Here is how rendered `-`, `--` and `---`:
image

Now it is `--`, would you like `---` ?

File: docs/gradient-checkpointing.md [hunk 1]
```diff
@@ -205,7 +205,7 @@ Using words, this alternative implementation doesn't compute `g_vjp`, or the res
 
 The cost you pay is redundant work: in `f_bwd2` you must re-evaluate `g(x)` as part of `jax.vjp(g, x)` just to discard its value (in the underscore variable on the line `_, g_vjp = jax.vjp(g, x)`).
 
-You can get this VJP behavior in autodiff &#151; without having to write VJP functions directly &#151; by instead using {func}`jax.checkpoint` in an alternative definition of the original function `f`:
+You can get this VJP behavior in autodiff -- without having to write VJP functions directly -- by instead using {func}`jax.checkpoint` in an alternative definition of the original function `f`:
```
Comment (vfdev-5) [inline docs/gradient-checkpointing.md]:
```suggestion
You can get this VJP behavior in autodiff --- without having to write VJP functions directly --- by instead using {func}`jax.checkpoint` in an alternative definition of the original function `f`:
```

File: docs/notebooks/autodiff_remat.ipynb [hunk 2]
```diff
@@ -473,7 +473,7 @@
     "id": "LqTrjPoGqrK7"
    },
    "source": [
-    "We can get this VJP behavior in autodiff &#151; without having to write VJP functions directly &#151; by instead using `jax.checkpoint` in an alternative definition of the original function `f`:"
+    "We can get this VJP behavior in autodiff -- without having to write VJP functions directly -- by instead using `jax.checkpoint` in an alternative definition of the original function `f`:"
```
Comment (vfdev-5) [inline docs/notebooks/autodiff_remat.ipynb]:
```suggestion
    "We can get this VJP behavior in autodiff --- without having to write VJP functions directly --- by instead using `jax.checkpoint` in an alternative definition of the original function `f`:"
```

**Comment by hawkinsp** [comment]
Yes please, `---` is the right dash to use here.

`--` is used only for number ranges.

File: docs/notebooks/autodiff_remat.md [hunk 3]
```diff
@@ -231,7 +231,7 @@ The cost we pay is redundant work: in `f_bwd2` we must re-evaluate `g(x)` as par
 
 +++ {"id": "LqTrjPoGqrK7"}
 
-We can get this VJP behavior in autodiff &#151; without having to write VJP functions directly &#151; by instead using `jax.checkpoint` in an alternative definition of the original function `f`:
+We can get this VJP behavior in autodiff -- without having to write VJP functions directly -- by instead using `jax.checkpoint` in an alternative definition of the original function `f`:
```
Comment (vfdev-5) [inline docs/notebooks/autodiff_remat.md]:
```suggestion
We can get this VJP behavior in autodiff --- without having to write VJP functions directly --- by instead using `jax.checkpoint` in an alternative definition of the original function `f`:
```

**Review by hawkinsp — APPROVED** [review]
